src/lib.rs

```diff
@@ -0,0 +1,66 @@
+use std::io;
+
+pub struct Config {
+    pub version: String,
+}
+
+impl Config {
+    pub fn new() -> Config {
+        let version = format!(
+            "{} {}",
+            env!("CARGO_PKG_NAME").to_string(),
+            env!("CARGO_PKG_VERSION").to_string()
+        );
+
+        Config { version }
+    }
+}
+
+pub fn run(config: Config, stdout: &mut io::Write) {
+    let version_string = format!("{}\n", config.version);
+
+    stdout
+        .write(version_string.as_bytes())
+        .expect("Something went wrong.");
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    struct StdoutDouble {
+        pub written_content: String,
+    }
+
+    impl StdoutDouble {
+        fn new() -> StdoutDouble {
+            let version = "";
+
+            StdoutDouble {
+                written_content: version.to_string(),
+            }
+        }
+    }
+
+    impl io::Write for StdoutDouble {
+        fn write(&mut self, _content: &[u8]) -> std::result::Result<usize, std::io::Error> {
+            self.written_content = std::str::from_utf8(_content).unwrap().to_string();
+
+            Ok(0)
+        }
+        fn flush(&mut self) -> std::result::Result<(), std::io::Error> {
+            Ok(())
+        }
+    }
+
+    #[test]
+    fn writes_version_to_output() {
+        let mut stdout = StdoutDouble::new();
+        let mut config = Config::new();
+        config.version = String::from("the version");
```

Comment:
alternative: `"the version".into()`

Reply:
Didn't know about `into()` before, cheers!